mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -30,52 +30,83 @@ use mm2_metrics::MetricsOps;
 use mm2_number::{construct_detailed, BigDecimal};
 use serde_json::{self as json, Value as Json};
 use std::borrow::Cow;
+use std::iter::Extend;
 
 use crate::mm2::lp_dispatcher::{dispatch_lp_event, StopCtxEvent};
 use crate::mm2::lp_network::subscribe_to_topic;
-use crate::mm2::lp_ordermatch::{cancel_orders_by, CancelBy};
-use crate::mm2::lp_swap::{active_swaps_using_coin, tx_helper_topic, watcher_topic};
+use crate::mm2::lp_ordermatch::{cancel_orders_by, get_matching_orders, CancelBy};
+use crate::mm2::lp_swap::{active_swaps_using_coins, tx_helper_topic, watcher_topic};
 use crate::mm2::MmVersionResult;
 
+const INTERNAL_SERVER_ERROR_CODE: u16 = 500;
+const RESPONSE_OK_STATUS_CODE: u16 = 200;
+
 /// Attempts to disable the coin
 pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
     let ticker = try_s!(req["coin"].as_str().ok_or("No 'coin' field")).to_owned();
-    let _coin = match lp_coinfind(&ctx, &ticker).await {
+    let coin = match lp_coinfind(&ctx, &ticker).await {
         Ok(Some(t)) => t,
         Ok(None) => return ERR!("No such coin: {}", ticker),
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
-    let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
-    }
-    let (cancelled, still_matching) = try_s!(cancel_orders_by(&ctx, CancelBy::Coin { ticker: ticker.clone() }).await);
-    if !still_matching.is_empty() {
+    let coins_ctx = try_s!(CoinsContext::from_ctx(&ctx));
+
+    // If a platform coin is to be disabled, we get all the enabled tokens for this platform coin first.
+    let mut coins_to_disable = coins_ctx.get_tokens_to_disable(&ticker).await;
+    // We then add the platform coin to the list of the coins to be disabled.
+    coins_to_disable.insert(ticker.clone());
+    drop_mutability!(coins_to_disable);
+
+    // Get all matching orders and active swaps.
+    let active_swaps = try_s!(active_swaps_using_coins(&ctx, &coins_to_disable));
+    let still_matching_orders = try_s!(get_matching_orders(&ctx, &coins_to_disable).await);
+
+    // If there're matching orders or active swaps we return an error.
+    if !active_swaps.is_empty() || !still_matching_orders.is_empty() {
         let err = json!({
-            "error": format!("There're currently matching orders using {}", ticker),
-            "orders": {
-                "matching": still_matching,
-                "cancelled": cancelled,
-            }
+            "error": format!("There're currently matching orders or active swaps for some tokens"),
+            "active_swaps": active_swaps,
+            "still_matching": still_matching_orders
```

Comment:
I think the error answer should be:
```json
{
  "error": "ERR",
  "orders": {
    "matching": still_matching,
    "cancelled": cancelled,
  },
  "active_swaps": active_swaps
}
```
And this JSON structure should be used on any error since `if !active_swaps.is_empty() || !still_matching_orders.is_empty() {` line.

You can add the following helper structure:
```rust
#[derive(Serialize)]
struct DisableError {
  error: String,
  orders: DisableErrorOrdersInfo, // also this strucutre
  active_swaps: Vec<_>,
}
```
They can be declared inside or outside of `disable_coin` function

Reply:
I think this is better 
```rust
#[derive(Serialize)]
pub struct DisableCoinError;

impl DisableCoinError {
    pub fn active_or_still_matching_err(
        error: String,
        matching: &[Uuid],
        canclled: &[String],
        active_swaps: &[Uuid],
    ) -> Result<Response<Vec<u8>>, String> {
        let err = json!({
            "error": error,
            "orders": {
                "matching": matching,
                "canclled": canclled
            },
            "active_swaps": active_swaps
        });
        Response::builder()
            .status(INTERNAL_SERVER_ERROR_CODE)
            .body(json::to_vec(&err).unwrap())
            .map_err(|e| ERRL!("{}", e))
    }
}

pub fn disabled_err(
        error: String,
        cancelled_orders: &[Uuid],
        disabled_tokens: &[String],
    ) -> Result<Response<Vec<u8>>, String> {
        let err = json!({
            "error": error,
            "cancelled_orders": cancelled_orders,
            "disabled_tokens": disabled_tokens,
        });
        Response::builder()
            .status(INTERNAL_SERVER_ERROR_CODE)
            .body(json::to_vec(&err).unwrap())
            .map_err(|e| ERRL!("{}", e))
    }
```

Reply:
Yes, LGTM! Only this
```rust
let err = json!({
            "error": error,
            "cancelled_orders": cancelled_orders,
            "disabled_tokens": disabled_tokens,
        });
```
should have the same fields as in `active_or_still_matching_err`, because if an error occurred on `coins_ctx.remove_coin`, we can expect that we didn't actually return any coin

Reply:
And `derive(Serialize)` is not required anymore